pkg/core/native_neo_test.go

```diff
@@ -38,203 +27,6 @@ func checkTxHalt(t *testing.T, bc *Blockchain, h util.Uint256) {
-			require.NoError(t, err)
-			newGAS.Sub(newGAS, gasForHold)
-			require.True(t, newGAS.Sign() > 0)
-			gasBalance[i] = newGAS
-		}
-		// First account voted later than the others.
-		require.Equal(t, -1, gasBalance[0].Cmp(gasBalance[1]))
-		for i := 2; i < testchain.ValidatorsCount; i++ {
-			require.Equal(t, 0, gasBalance[i].Cmp(gasBalance[1]))
-		}
-		require.Equal(t, 1, gasBalance[1].Cmp(gasBalance[testchain.ValidatorsCount]))
-		for i := testchain.ValidatorsCount; i < testchain.CommitteeSize(); i++ {
-			require.Equal(t, 0, gasBalance[i].Cmp(gasBalance[testchain.ValidatorsCount]))
-		}
-	})
-
-	require.NoError(t, neo.UnregisterCandidateInternal(ic, candidates[0]))
-	require.Error(t, neo.VoteInternal(ic, h, candidates[0]))
-	advanceChain(t)
-
-	pubs, err = neo.ComputeNextBlockValidators(bc, ic.DAO)
-	require.NoError(t, err)
-	for i := range pubs {
-		require.NotEqual(t, candidates[0], pubs[i])
-	}
-}
-
-// TestNEO_RecursiveDistribution is a test for https://github.com/nspcc-dev/neo-go/pull/2181.
-func TestNEO_RecursiveGASMint(t *testing.T) {
-	bc := newTestChain(t)
-	initBasicChain(t, bc)
-
-	contractHash, err := bc.GetContractScriptHash(1) // deployed rpc/server/testdata/test_contract.go contract
-	require.NoError(t, err)
-	tx := transferTokenFromMultisigAccount(t, bc, contractHash, bc.contracts.GAS.Hash, 2_0000_0000)
-	checkTxHalt(t, bc, tx.Hash())
-
-	// Transfer 10 NEO to test contract, the contract should earn some GAS by owning this NEO.
-	tx = transferTokenFromMultisigAccount(t, bc, contractHash, bc.contracts.NEO.Hash, 10)
-	res, err := bc.GetAppExecResults(tx.Hash(), trigger.Application)
-	require.NoError(t, err)
-	require.Equal(t, vm.HaltState, res[0].VMState)
-
-	// Add blocks to be able to trigger NEO transfer from contract address to owner
-	// address inside onNEP17Payment (the contract starts NEO transfers from chain height = 100).
-	for i := bc.BlockHeight(); i < 100; i++ {
-		require.NoError(t, bc.AddBlock(bc.newBlock()))
-	}
-
-	// Transfer 1 more NEO to the contract. Transfer will trigger onNEP17Payment. OnNEP17Payment will
-	// trigger transfer of 11 NEO to the contract owner (based on the contract code). 11 NEO Transfer will
-	// trigger GAS distribution. GAS transfer will trigger OnNEP17Payment one more time. The recursion
-	// shouldn't occur here, because contract's balance LastUpdated height has already been updated in
-	// this block.
-	tx = transferTokenFromMultisigAccount(t, bc, contractHash, bc.contracts.NEO.Hash, 1)
-	res, err = bc.GetAppExecResults(tx.Hash(), trigger.Application)
-	require.NoError(t, err)
-	require.Equal(t, vm.HaltState, res[0].VMState, res[0].FaultException)
-}
-
-func TestNEO_SetGasPerBlock(t *testing.T) {
-	bc := newTestChain(t)
-
-	testGetSet(t, bc, bc.contracts.NEO.Hash, "GasPerBlock",
-		5*native.GASFactor, 0, 10*native.GASFactor)
-}
-
 func TestNEO_CalculateBonus(t *testing.T) {
```

Comment:
It's not moved because of `CalculateNEOHolderReward` also? You can do claims and check GAS balance then.

Reply:
It is not moved because it tests the negative NEO balance error handler (this situation can't be reproduced with *fair* neotest framework). Although it's one line only and it hardly ever possible to get negative balance there, so I've moved this test also. Fixed.